backend/lambdas/jobs/stats_updater.py

```diff
@@ -30,82 +33,86 @@ def _aggregate_stats(events):
-                {
-                    "TotalQueryCount": 1,
-                    "TotalQuerySucceededCount": 1
-                    if event_name == "QuerySucceeded"
-                    else 0,
-                    "TotalQueryFailedCount": 1 if event_name == "QueryFailed" else 0,
-                    "TotalQueryScannedInBytes": event_data.get("Statistics", {}).get(
-                        "DataScannedInBytes", 0
-                    ),
-                    "TotalQueryTimeInMillis": event_data.get("Statistics", {}).get(
-                        "EngineExecutionTimeInMillis", 0
-                    ),
-                }
-            )
-        if event_name in [
-            "ObjectUpdated",
-            "ObjectUpdateFailed",
-            "ObjectRollbackFailed",
-        ]:
-            stats += Counter(
-                {
-                    "TotalObjectUpdatedCount": 1
-                    if event_name == "ObjectUpdated"
-                    else 0,
-                    "TotalObjectUpdateFailedCount": 1
-                    if event_name == "ObjectUpdateFailed"
-                    else 0,
-                    "TotalObjectRollbackFailedCount": 1
-                    if event_name == "ObjectRollbackFailed"
-                    else 0,
-                }
-            )
+            stats += Counter({
+                "TotalQueryCount": 1,
+                "TotalQuerySucceededCount": 1 if event_name == "QuerySucceeded" else 0,
+                "TotalQueryFailedCount": 1 if event_name == "QueryFailed" else 0,
+                "TotalQueryScannedInBytes": event_data.get("Statistics", {}).get("DataScannedInBytes", 0),
+                "TotalQueryTimeInMillis": event_data.get("Statistics", {}).get("EngineExecutionTimeInMillis", 0),
+            })
+        if event_name in ["ObjectUpdated", "ObjectUpdateFailed", "ObjectRollbackFailed"]:
+            stats += Counter({
+                "TotalObjectUpdatedCount": 1 if event_name == "ObjectUpdated" else 0,
+                "TotalObjectUpdateFailedCount": 1 if event_name == "ObjectUpdateFailed" else 0,
+                "TotalObjectRollbackFailedCount": 1 if event_name == "ObjectRollbackFailed" else 0,
+            })
+            if event_name == "ObjectUpdateFailed":
+                # TODO:: this can only work when deleteOldVersion flag is True for all involved data mappers
+                try:
+                    data_access_role_arn = event_data["Message"].get("RoleArn", getenv("DataAccessRoleArn"))
+                    session = get_session(data_access_role_arn)
+                    client = session.client("s3")
+                    s3_object = event_data["Message"]["Object"]
+                    input_bucket, input_key = parse_s3_url(s3_object)
+                    revert_last(client, input_bucket, input_key)
+                except Exception:
+                    logger.exception("Unable to revert last")
+                    try:
+                        purge_queue(deletion_queue_url)
```

Comment:
Why purging the queue in case of error? I'm curious. You prefer to quit as soon as you have an error and restart a job rather than keep trying for other objects first?

Reply:
As described in the reply above, ObjectUpdateFailed may indicate file corruption,
hence in case the revert failed we want to stop the job and check if the data is corrupted.
if it is, we need to locate the corrupted file and revert it manually.
This state is not expected to happen therefore assuming it did we rather stop and find the cause in order to prevent the corruptions of further files.